packages/test/mocha-test-setup/src/mochaHooks.ts

```diff
@@ -24,112 +24,119 @@ const propsDict =
-		// end of a test, in case the testName is undefined, we will use the currentTestName.
-		event.testName = this.testName ?? currentTestName;
+		if (this.currentTestName !== undefined) {
+			event.testName = this.currentTestName;
+		}
 		event.testVariant = testVariant;
-		event.hostName = pkgName;
 		this.parentLogger.send({
 			...event,
-			// If there's an override for the hostName in FLUID_LOGGER_PROPS,
-			// display that in the telemetry instead of the package name.
+			// Setting hostname to pkgName is the behavior we had for a long time, so keeping it just in case.
+			// But prefer a value set through FLUID_LOGGER_PROPS if it exists.
 			hostName: propsDict?.hostName ?? pkgName,
 			details: JSON.stringify(propsDict),
 		});
 	}
 	async flush() {
 		return this.parentLogger.flush();
 	}
-	constructor(
-		private readonly parentLogger: ITelemetryBufferedLogger,
-		private readonly testName: string | undefined,
-	) {}
+	constructor(private readonly parentLogger: ITelemetryBufferedLogger) {}
+
+	/**
+	 * Sets the test that is currently running.
+	 * The test name will be included in all events logged through the logger until {@link clearCurrentTest} is called.
+	 * @param testName - The name of the test that is currently running.
+	 */
+	public setCurrentTest(testName: string) {
+		this.currentTestName = testName;
+	}
+
+	/**
+	 * Clears the test that is currently running.
+	 * Must be called after a given test has completed so that events logged outside the context of a test
+	 * don't include the name of the last test that ran.
+	 */
+	public clearCurrentTest() {
+		this.currentTestName = undefined;
+	}
 }
 const nullLogger: ITelemetryBufferedLogger = {
 	send: () => {},
 	flush: async () => {},
 };
 
+// Keep references to the original console functions so we can restore them after each test.
 const log = console.log;
 const error = console.log;
 const warn = console.warn;
-let currentTestLogger: ITelemetryBufferedLogger | undefined;
-let currentTestName: string | undefined;
-let originalLogger: ITelemetryBufferedLogger;
+
+const originalLogger = _global.getTestLogger?.() ?? nullLogger;
+const testLogger = new FluidTestRunLogger(originalLogger);
 
 /**
  * @internal
  */
 export const mochaHooks = {
 	beforeAll() {
-		originalLogger = _global.getTestLogger?.() ?? nullLogger;
-		_global.getTestLogger = () => {
-			// If a current test logger exists, use that. Otherwise, create a new one. This should become the
-			// current test logger if this function is running in a context which understands the current test.
-			// Otherwise, just return the created TestLogger. (This happens e.g. if someone calls `getTestLogger`
-			// in a `before` or `after` hook, due to the order in which mocha hooks run)
-			if (currentTestLogger !== undefined) {
-				return currentTestLogger;
-			}
-
-			const testLogger = new TestLogger(originalLogger, currentTestName);
-			if (currentTestName !== undefined) {
-				currentTestLogger = testLogger;
-			}
```

Comment:
One notable change is that we don't instantiate a new logger for each test (or for each call to `getTestLogger` outside the context of a running test) anymore. But the way we were doing it before felt very messy (`beforeEach`/`afterEach` set/cleared the test name, and depending on that, this function returned an existing logger or created a new one and "cached it" as the currentTestLogger or created a new one and didn't cache it. I can also see an approach where we keep instantiating a new logger per test but do it more cleanly than this; but the approach I took of a single logger with `setCurrentTest`/`clearCurrentTest` felt ok.